lib/googleauth/credentials.rb

```diff
@@ -40,10 +40,55 @@ module Auth
     class Credentials
       TOKEN_CREDENTIAL_URI = "https://oauth2.googleapis.com/token".freeze
```

Comment:
Are these values intended to be changed? When we originally created `Credentials` it was an internal class, so we didn't worry about how users would interact with it as it wasn't something intended to be interacted with.

If we expect users to change `TOKEN_CREDENTIAL_URI` or `AUDIENCE`, then we should probably add `token_credential_uri` and `audience` methods. However, we will want to keep the constants to define the default value for them.

Reply:
If we don't expect users to customize these values, then I think we should keep them as constants.

Reply:
I believe they should be customizable

Reply:
Added.